apps/condo/domains/banking/schema/BankAccount.js

```diff
@@ -0,0 +1,195 @@
+
+                    const country = get(newItem, 'country')
+                    const number = get(newItem, 'number')
+                    const routingNumber = get(newItem, 'routingNumber')
+
+                    const { result, errors } = validateNumber(number, routingNumber, country)
+
+                    if ( !result ) {
+                        addFieldValidationError(errors[0])
+                    }
+                },
+            },
+        },
+
+        currencyCode: CURRENCY_CODE_FIELD,
+
+        approvedAt: {
+            schemaDoc: 'When the bank account received the status of approved',
+            type: DateTimeUtc,
+            isRequired: false,
+            access: {
+                read: true,
+                create: access.canManageIsApprovedField,
+                update: access.canManageIsApprovedField,
+            },
+        },
+
+        approvedBy: {
+            schemaDoc: 'Who set the approved status for the bank account',
+            type: 'Relationship',
+            ref: 'User',
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+            access: {
+                read: true,
+                create: access.canManageIsApprovedField,
+                update: access.canManageIsApprovedField,
+            },
+        },
+
+        importId: IMPORT_ID_FIELD,
+
+        territoryCode: {
+            schemaDoc: 'Location code (Classifier of Territories of Municipal Units - OKTMO)',
+            type: Text,
+            isRequired: false,
+        },
+
+        bankName: {
+            schemaDoc: 'Bank name',
+            type: Text,
+            isRequired: false,
+        },
+
+        meta: {
+            schemaDoc: 'Structured metadata',
+            type: Json,
+            isRequired: false,
+        },
+
+        tinMeta: {
+            schemaDoc: 'Structured metadata found by tin',
+            type: Json,
+            isRequired: false,
+        },
+
+        routingNumberMeta: {
+            schemaDoc: 'Structured metadata found by routing number',
+            type: Json,
+            isRequired: false,
+        },
+
+    },
+    plugins: [uuided(), versioned(), tracked(), softDeleted(), dvAndSender(), historical()],
+    access: {
+        read: access.canReadBankAccounts,
+        create: access.canManageBankAccounts,
+        update: access.canManageBankAccounts,
+        delete: false,
+        auth: true,
+    },
+    hooks: {
+        resolveInput: async ({ resolvedData, context }) => {
+
+            // If bank account is being updated -> drop approvedBy and approvedAt!
+            if (('approvedAt' in resolvedData && get(resolvedData, 'approvedAt'))
+                || ('approvedBy' in resolvedData && get(resolvedData, 'approvedBy'))) {
+                const dateNow = new Date().toISOString()
+                const { authedItem: { id = null } = {} } = context
```

Comment:
It seems like there will be no case when `resolveInput` will be invoked for unauthenticated user.
Otherwise, when it will happen, you will get `null` in `approvedBy` and today timestamp in `approvedAt`

```suggestion
                const { authedItem: { id = null } = {} } = context
```